task 3

Artem197817 opened the pull request:
Третье задание

Artem197817 committed:
task 3

jsru-1 commented:
Добавляю преподавателя (@ufocoder) для код-ревью.

ufocoder commented on 03-objects-arrays-intro-to-testing/1-create-getter/index.js:
```diff
@@ -4,5 +4,19 @@
+        let current = obj;
+        for (let i = 0; i < parts.length; i++) {
+            if (current === null || current === undefined || !Object.prototype.hasOwnProperty.call(current, parts[i])) {
+                return undefined;
```
`return undefined;` -> `return;`

ufocoder commented on 03-objects-arrays-intro-to-testing/2-invert-object/index.js:
```diff
@@ -4,5 +4,16 @@
 export function invertObj(obj) {
-
+    if (!obj || typeof obj !== 'object') {
+        return undefined;
```
`return undefined;` -> `return;`

ufocoder commented on 03-objects-arrays-intro-to-testing/2-invert-object/index.js:
```diff
@@ -4,5 +4,16 @@
+    }
+    const inverted = {};
+    for (const key in obj) {
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
```
В чем необходимость `hasOwnProperty` ?

ufocoder commented on 03-objects-arrays-intro-to-testing/1-create-getter/index.js:
```diff
@@ -4,5 +4,19 @@
+    return function (obj) {
+        let current = obj;
+        for (let i = 0; i < parts.length; i++) {
+            if (current === null || current === undefined || !Object.prototype.hasOwnProperty.call(current, parts[i])) {
```
Вместо множества проверок можно использовать 1:
```
if (current.?hasOwnProperty(parts[i]))
```

Artem197817 commented:
В первых двух комментариях не понятна запись return undefined; -> return;До этого изучал Java  теперь не  все конструкции JavaScript понятны еще hasOwnProperty необходимо для проверки что свойство является собственным свойством объекта а не наследуется В этом случае конечно можно обойтись без него По четвертому понятно. А вообще пока сложно достаточно. Иногда сначала проще решить задачу на Java если это возможно  потом перевести на  JavaScriptВ джаве как-то все более структурировано здесь больше творчества и нюансов.(Условно в джаве одни кавычки) и один метод определения функции), но интересно. 10.06.2024, 18:51, "Sergey" ***@***.***>***@***.*** commented on this pull request.In 03-objects-arrays-intro-to-testing/1-create-getter/index.js:> @@ -4,5 +4,19 @@
  * @returns {function} - function-getter which allow get value from object by set path
  */
 export function createGetter(path) {
+    const parts = path.split('.');
+    return function (obj) {
+        let current = obj;
+        for (let i = 0; i < parts.length; i++) {
+            if (current === null || current === undefined || !Object.prototype.hasOwnProperty.call(current, parts[i])) {
+                return undefined;
return undefined; -> return;In 03-objects-arrays-intro-to-testing/2-invert-object/index.js:> @@ -4,5 +4,16 @@
  * @returns {object | undefined} - returns new object or undefined if nothing did't pass
  */
 export function invertObj(obj) {
-
+    if (!obj || typeof obj !== 'object') {
+        return undefined;
return undefined; -> return;In 03-objects-arrays-intro-to-testing/2-invert-object/index.js:> @@ -4,5 +4,16 @@
  * @returns {object | undefined} - returns new object or undefined if nothing did't pass
  */
 export function invertObj(obj) {
-
+    if (!obj || typeof obj !== 'object') {
+        return undefined;
+    }
+    const inverted = {};
+    for (const key in obj) {
+        if (Object.prototype.hasOwnProperty.call(obj, key)) {
В чем необходимость hasOwnProperty ?In 03-objects-arrays-intro-to-testing/1-create-getter/index.js:> @@ -4,5 +4,19 @@
  * @returns {function} - function-getter which allow get value from object by set path
  */
 export function createGetter(path) {
+    const parts = path.split('.');
+    return function (obj) {
+        let current = obj;
+        for (let i = 0; i < parts.length; i++) {
+            if (current === null || current === undefined || !Object.prototype.hasOwnProperty.call(current, parts[i])) {
Вместо множества проверок можно использовать 1:if (current.?hasOwnProperty(parts[i]))
—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>  -- С уважением "Катод"  

Artem197817 commented:
В первых двух комментариях не понятна запись return undefined; -> return;
До этого изучал Java  теперь не  все конструкции JavaScript понятны еще
 
hasOwnProperty необходимо для проверки что свойство является собственным свойством объекта а не наследуется В этом случае конечно можно обойтись без него
 
По четвертому понятно.
 
А вообще пока сложно достаточно. Иногда сначала проще решить задачу на Java если это возможно  потом перевести на  JavaScript
В джаве как-то все более структурировано здесь больше творчества и нюансов.(Условно в джаве одни кавычки) и один метод определения функции), но интересно.
не дождался ответа по почте

ufocoder reviewed (approved):
Переходите к компонентам
